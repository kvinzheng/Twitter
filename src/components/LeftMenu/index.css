.LeftMenu {
  display: inline-block;
  width: 20%;
  vertical-align: top;
}

.LeftMenu-Nav {
  display: flex;
  flex-direction: column;
}

.LeftMenu_Nav-Link {
  width: 100%;
  color: black;
  font-size: 16px;
  font-weight: bold;
  padding: 4px 20px 4px 20px;
  text-align: left;
}

.LeftMenu_Nav-Link,
.LeftMenu_Nav-Link:hover {
  text-decoration: none;
}

.LeftMenu_Nav-Link .LeftMenu-Twitter-Logo {
  color: rgba(29, 161, 242, 1);
  border-color: rgba(29, 161, 242, 1);
}

.active {
  color: rgba(29, 161, 242, 1);
}

.LeftMenu-Link-Wrapper {
  display: inline-flex;
  border-radius: 50px;
  align-items: center;
  padding: 9px;
}

.LeftMenu-Link-Wrapper span {
  padding: 0 12px 0 10px;
}

.LeftMenu_Nav-Link:hover .LeftMenu-NavGroup {
  background: #cceeff;
}

.LeftMenu-TweetButton {
  padding: 0px 25px;
  border-radius: 50px;
  color: white;
  background-color: rgba(29, 161, 242, 1);
  box-shadow:rgba(0, 0, 0, 0.08) 0px 8px 28px;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 43px;
}

.LeftMenu-TweetButton:hover {
  background-color: #0088cc;
}
