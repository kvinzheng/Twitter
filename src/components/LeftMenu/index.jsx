import React from "react";
import {
  Home,
  Hash,
  Bell,
  Mail,
  List,
  Circle,
  MoreHorizontal,
  Twitter,
} from "react-feather";
import "./index.css";

export default function LeftMenu() {
  return (
    <div className="LeftMenu">
      <div className="LeftMenu-Nav">
        <a className="LeftMenu_Nav-Link" href="#">
          <div className="LeftMenu-Link-Wrapper">
            <Twitter
              size="30"
              fill="rgba(29,161,242,1.00)"
              className="LeftMenu-Twitter-Logo"
            />
          </div>
        </a>
        <a className="LeftMenu_Nav-Link active " href="#">
          <div className="LeftMenu-Link-Wrapper">
            <Home size="21" />
            <span>Home</span>
          </div>
        </a>
        <a className="LeftMenu_Nav-Link" href="#">
          <div className="LeftMenu-Link-Wrapper">
            <Hash size="21" />
            <span>Explore</span>
          </div>
        </a>
        <a className="LeftMenu_Nav-Link" href="#">
          <div className="LeftMenu-Link-Wrapper">
            <Bell size="21" />
            <span>Notifications</span>
          </div>
        </a>
        <a className="LeftMenu_Nav-Link" href="#">
          <div className="LeftMenu-Link-Wrapper">
            <Mail size="21" />
            <span>Messages</span>
          </div>
        </a>
        <a className="LeftMenu_Nav-Link" href="#">
          <div className="LeftMenu-Link-Wrapper">
            <List size="21" />
            <span>Lists</span>
          </div>
        </a>
        <a className="LeftMenu_Nav-Link" href="#">
          <div className="LeftMenu-Link-Wrapper">
            <Circle size="21" />
            <span>Profile</span>
          </div>
        </a>
        <a className="LeftMenu_Nav-Link" href="#">
          <div className="LeftMenu-Link-Wrapper">
            <MoreHorizontal />
            <span>More</span>
          </div>
        </a>
        <a className="LeftMenu_Nav-Link" href="#">
          <div className="LeftMenu-TweetButton">
            <span>Tweet</span>
          </div>
        </a>
      </div>
    </div>
  );
}
