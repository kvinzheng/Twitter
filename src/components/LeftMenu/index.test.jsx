import React from "react";

import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import LeftMenu from "./index";

configure({ adapter: new Adapter() });

describe("LeftMenu", () => {
  it("should render without any data defined", () => {
    const component = shallow(<LeftMenu />);

    expect(component).toMatchSnapshot();
  });
  it("Renders with correct classes", () => {
    const component = shallow(<LeftMenu />);
    expect(component.find(".LeftMenu").length).toBe(1);
    expect(component.find(".LeftMenu_Nav-Link").length).toBe(9);
  });
});
