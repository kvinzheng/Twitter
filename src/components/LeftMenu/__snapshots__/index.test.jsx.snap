// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftMenu should render without any data defined 1`] = `ShallowWrapper {}`;
